falcon_kit/stats_preassembly.py

```diff
@@ -115,7 +116,25 @@ def read_lens_from_db(db_fn):
     """
     return list(sorted(get_db_readlengths(db_fn)))
 
-def stats_dict(stats_raw_reads, stats_seed_reads, stats_corrected_reads, genome_length, length_cutoff):
+def metric_fragmentation(preads_dir):
+    # https://jira.pacificbiosciences.com/browse/SAT-105
+    bash = r"""
+sed -nr 's;>prolog/([0-9]*)[0-9]/.*;\1;p' %s/*.fasta | sort | uniq -c | awk '{print $1}' | sort | uniq -c
```

Comment:
The sed command itself is not what I'm thinking about, it's all the merging and sorting after the pipes that makes me cringe.  It's not slow on bacteria, but will be really slow on larger genomes.